src/daemon/daemonlocalserverconnection.cpp

```diff
@@ -91,6 +95,24 @@ void DaemonLocalServerConnection::parseCommand(const QByteArray& data) {
   QString type = typeValue.toString();
 
   logger.debug() << "Command received:" << type;
+  auto session = Daemon::instance()->session();
+  if (!session->isActive()) {
+    if (type == "activate" && !session->start(m_socket)) {
+      logger.error() << "Unable to start new session."
+                        "Ignoring activate command.";
+      return;
+      // Other than "activate", only "status" command is available for any user.
+    } else if (type != "status") {
+      logger.error() << "Attempted to send command:" << type
+                     << "but there is no ongoing session. Ignoring.";
+      return;
+    }
+  } else {
+    if (!session->isPeerAuthorized(m_socket)) {
+      logger.warning() << "Caller is unauthorized. Ignoring command.";
+      return;
+    }
+  }
```

Comment:
Consider extracting a method for this block.

Reply:
I moved this to the DaemonAccessControl class into it's own method.